### ISSUE TITLE
Broken Pango in Graphviz_jll

### ISSUE THREAD

**Issue by epatters** [JuliaPackaging/Yggdrasil#1428]
Migrated from https://github.com/AlgebraicJulia/Catlab.jl/pull/232: On my Mac, whenever I run `dot` from `Graphviz_jll`, I get the warning

```
Warning: Could not load "/Users/epatters/.julia/artifacts/6457d23222d6cb27ff900ad8d7ecaa0bfb2c4332/lib/graphviz/libgvplugin_pango.6.dylib" - It was found, so perhaps one of its dependents was not.  Try ldd.
```

All output formats that depend on Pango, including JPG, PNG, and PDF, are broken. However, the SVG and PS formats work fine.

```bash
epatters$ echo 'digraph g { n1 -> n2; }' | /Users/epatters/.julia/artifacts/6457d23222d6cb27ff900ad8d7ecaa0bfb2c4332/bin/dot -Tpdf
Warning: Could not load "/Users/epatters/.julia/artifacts/6457d23222d6cb27ff900ad8d7ecaa0bfb2c4332/lib/graphviz/libgvplugin_pango.6.dylib" - It was found, so perhaps one of its dependents was not.  Try ldd.
Warning: Could not load "/Users/epatters/.julia/artifacts/6457d23222d6cb27ff900ad8d7ecaa0bfb2c4332/lib/graphviz/libgvplugin_pango.6.dylib" - It was found, so perhaps one of its dependents was not.  Try ldd.
Format: "pdf" not recognized. Use one of: canon cmap cmapx cmapx_np dot dot_json eps fig gv imap imap_np ismap json json0 mp pdf pic plain plain-ext png pov ps ps2 svg svgz tk vdx vml vmlz xdot xdot1.2 xdot1.4 xdot_json
```

A similar problem was reported for Graphviz in conda-forge (https://github.com/conda-forge/graphviz-feedstock/issues/18) and later fixed (https://github.com/conda-forge/graphviz-feedstock/pull/23).

**Comment by isaacsas** [comment]
This seems to also be an issue when trying to use Graphviz_jll on Github actions, so this doesn't seem Mac specific. I get a similar error:
```
Format: "png" not recognized. Use one of: canon cmap cmapx cmapx_np dot dot_json eps fig gv imap imap_np ismap json json0 mp pic plain plain-ext pov ps ps2 svg svgz tk vdx vml vmlz xdot xdot1.2 xdot1.4 xdot_json
here!!!Graphs: Error During Test at /home/runner/.julia/packages/SafeTestsets/A83XK/src/SafeTestsets.jl:25
  Got exception outside of a @test
  LoadError: failed process: Process(`/home/runner/.julia/artifacts/edb465e7f0fe4ebd11efbafb9df31084e544ac07/bin/dot -Tpng`, ProcessExited(1)) [1]
```

**Comment by jpfairbanks** [comment]
It looks like you could use eps or svg and get vector graphics in there. That is probably good enough for most graphviz use cases. Vector graphics for life.

**Comment by ViralBShah** [comment]
@giordano seems to usually know the GTK stack details. Any suggestions here?

**Comment by giordano** [comment]
Did anyone try to run `ldd` as suggested by the warning?

**Comment by ViralBShah** [comment]
I don't see the file at all. Maybe not being built?

```
viralbshah@antarctic:~/.julia/artifacts/edb465e7f0fe4ebd11efbafb9df31084e544ac07/lib/graphviz$ ls
libgvplugin_core.la              libgvplugin_neato_layout.la
libgvplugin_core.so              libgvplugin_neato_layout.so
libgvplugin_core.so.6            libgvplugin_neato_layout.so.6
libgvplugin_core.so.6.0.0        libgvplugin_neato_layout.so.6.0.0
libgvplugin_dot_layout.la        libgvplugin_visio.la
libgvplugin_dot_layout.so        libgvplugin_visio.so
libgvplugin_dot_layout.so.6      libgvplugin_visio.so.6
libgvplugin_dot_layout.so.6.0.0  libgvplugin_visio.so.6.0.0
```

**Comment by ViralBShah** [comment]
Also nice to add the experimental platforms as well, if we are bumping versions as part of fixing this.

**Comment by giordano** [comment]
> I don't see the file at all. Maybe not being built?

It's in the macOS tarball:
```console
% tar tzvf Graphviz.v2.42.3.x86_64-apple-darwin14-cxx11.tar.gz | grep lib/graphviz/libgvplugin_pango.6.dylib
-r-xr-xr-x root/root     44228 2020-08-05 19:23 ./lib/graphviz/libgvplugin_pango.6.dylib
```

> Also nice to add the experimental platforms as well, if we are bumping versions as part of fixing this.

Quite frankly, Graphviz was particularly painful already the first time #351  I wasn't looking forward to a second time :smile:

**Comment by giordano** [comment]
@epatters

> whenever I run dot from Graphviz_jll

_How_ do you run `dot`?

**Comment by isaacsas** [comment]
I think the relevant code starts here: 

https://github.com/AlgebraicJulia/Catlab.jl/blob/8eb38a9afc4d6e35c9d48c6e1d7345091b1eb007/src/graphics/Graphviz.jl#L128

**Comment by giordano** [comment]
Executables should be run in this way: https://docs.binarybuilder.org/stable/jll/#ExecutableProduct (`Graphivz_jll` is a bit old, only the second syntax is available).  The documentation also explains why you have to do that.  There are good chances that would help.

**Comment by giordano** [comment]
Ok, looking at the configuration log I can confirm Graphviz on Linux was built without support for pangocairo, while it worked on macOS, reason why PDF works for @isaacsas in https://github.com/SciML/Catalyst.jl/pull/334. I have no clue about what happened in the linux build, I'd need to rebuild it, but honestly I don't have much time (not much intention) to enter again this painful rabbit hole.

**Comment by isaacsas** [comment]
Ok, in either case thanks for taking a look.

**Comment by Keno** [comment]
By setting
```
$ export DYLD_PRINT_LIBRARIES=1
$ export DYLD_PRINT_LIBRARIES_POST_LAUNCH=1
$ export DYLD_PRINT_RPATHS=1
```
we see
```
RPATH failed to expanding     @rpath/libbz2.1.0.6.dylib to: /Users/kfischer/.julia/artifacts/6457d23222d6cb27ff900ad8d7ecaa0bfb2c4332/lib/graphviz/../libbz2.1.0.6.dylib
RPATH failed to expanding     @rpath/libbz2.1.0.6.dylib to: /Users/kfischer/.julia/artifacts/6457d23222d6cb27ff900ad8d7ecaa0bfb2c4332/lib/./libbz2.1.0.6.dylib
RPATH failed to expanding     @rpath/libbz2.1.0.6.dylib to: /Users/kfischer/.julia/artifacts/6457d23222d6cb27ff900ad8d7ecaa0bfb2c4332/bin/../lib/libbz2.1.0.6.dylib
dyld: unloaded: /Users/kfischer/.julia/artifacts/6457d23222d6cb27ff900ad8d7ecaa0bfb2c4332/lib/graphviz/libgvplugin_pango.6.dylib
```
so the real issue is that it's missing the path to Bzip2. I'll include a fix in my graphviz bump PR.

**Comment by Keno** [comment]
Actually, the issue is just that it's looking for libbz2 at version 1.0.6, but the version we have is 1.0.8. So a simple rebuild should fix it. I guess the other dependencies don't use the SONAME in the same way.

**Comment by giordano** [comment]
Looks like this has been fixed by #3903, thanks Keno!

![image](https://user-images.githubusercontent.com/765740/142418644-e890ead5-a4ab-4464-be5b-61c88b45b624.png)

**Comment by giordano** [comment]
> Actually, the issue is just that it's looking for libbz2 at version 1.0.6, but the version we have is 1.0.8. So a simple rebuild should fix it. I guess the other dependencies don't use the SONAME in the same way.

Yeah, Bzip2 is... complicated, see #2957 and #2981, but from version 1.0.7 it should be all right everywhere.
